### PR TITLE
Fixes for Elasticsearch issue #184 -  Testing Elasticsearch functionality when network.host is not localhost.

### PR DESCRIPTION
**Pull request by rshad**
Hi team!

In this PR, we resolved the error reported in the issue https://github.com/wazuh/wazuh-ansible/issues/184.

**As @kravietz reported**

> The test in roles/wazuh-ansible/roles/elastic-stack/ansible-elasticsearch/tasks/main.yml line 87 only works if the elasticsearch_network_host is set to localhost but ElasticSearch uses this parameter to configure its bind address, so in many real-world instances it will be set to 0.0.0.0 or ::, in which case the test fails.

> - name: Make sure Elasticsearch is running before proceeding
  wait_for: host={{ elasticsearch_network_host }} port={{ elasticsearch_http_port }} delay=3 timeout=300
  tags:
    - configure
    - init  

To solve such issue it's needed to have 2 different variables, 1 for network.host and one for representing Elasticsearch host. In this case, we now have:

- `elasticsearch_network_host: 0.0.0.0` : Indicates the hosts which Elasticsearch API will accepts requests from.

>**network.host**
By default, Elasticsearch binds to loopback addresses only — e.g. 127.0.0.1 and [::1]. This is sufficient to run a single development node on a serve
- `elasticsearch_host: 127.0.0.1` : By now, it's used to check if Elasticsearch is running by checking if 127.0.0.1:9200 is available.

https://github.com/wazuh/wazuh-ansible/blob/fcb584ab2016a9d0e867b6748e1b13877067e3c8/roles/elastic-stack/ansible-elasticsearch/tasks/main.yml#L261-L265

https://github.com/wazuh/wazuh-ansible/blob/fcb584ab2016a9d0e867b6748e1b13877067e3c8/roles/elastic-stack/ansible-elasticsearch/defaults/main.yml#L4-L6

In this case, we set `elasticsearch_host` to `127.0.0.1` instead of  `0.0.0.0`  because `0.0.0.0` will not be considered as a valid IP, but will refer to the big-world domain. 

We also fixed some Ansible-Linting errors by refactoring Ansible tasks in the role of [`Elasticsearch`](https://github.com/wazuh/wazuh-ansible/tree/fcb584ab2016a9d0e867b6748e1b13877067e3c8/roles/elastic-stack/ansible-elasticsearch).

Kind regards,

Rshad

### CODE REVIEW

**Comment by manuasir** [comment]
Please @rshad , change the PR title.

File: roles/elastic-stack/ansible-elasticsearch/defaults/main.yml [hunk 1]
```diff
@@ -2,7 +2,8 @@
 elasticsearch_cluster_name: wazuh
 elasticsearch_node_name: node-1
 elasticsearch_http_port: 9200
-elasticsearch_network_host: 127.0.0.1
+elasticsearch_network_host: 0.0.0.0
```
Comment (manuasir) [inline roles/elastic-stack/ansible-elasticsearch/defaults/main.yml]: I think we should set this default value as `127.0.0.1`

File: roles/elastic-stack/ansible-elasticsearch/tasks/Debian.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@
 
 - name: Install Oracle Java 8
   become: true
-  apt: name=openjdk-8-jdk state=latest
+  apt: name=openjdk-8-jdk
```
Comment (manuasir) [inline roles/elastic-stack/ansible-elasticsearch/tasks/Debian.yml]: Why is the `state=latest` statement not needed anymore?

Comment (rshad) [reply roles/elastic-stack/ansible-elasticsearch/tasks/Debian.yml]: Hi @manuasir,

This change came as result to Ansible-Linting error:

` Package installs should not use latest ` reported in this issue: https://github.com/ansible/ansible-lint/issues/479

This change was made in this commit: https://github.com/ansible/ansible-lint/commit/f5fcee4eb3bea623e625de2ede02a3e2686e5e1e#diff-42252eea388ddb7d54414f584e9ebc78.

Sincerely, and like many other users facing this issue, I think this restriction has no logic due to the need to upgrade a package to its latest version without worrying about specifying a determined one. 

Anyway, I think `apt` will take the last version by default, so no worries.

Kind regards,

Rshad

File: roles/elastic-stack/ansible-elasticsearch/defaults/main.yml (hunk 1 above)
Comment (rshad) [reply roles/elastic-stack/ansible-elasticsearch/defaults/main.yml]: Hi @manuasir ,

No problem. We set it to 0.0.0.0 to simulate the error reported in the related issue.

I've just changed its value to `127.0.0.1` in https://github.com/wazuh/wazuh-ansible/commit/125b2114e6c77f089aac16bf4668cde2290efcbc .

https://github.com/wazuh/wazuh-ansible/blob/125b2114e6c77f089aac16bf4668cde2290efcbc/roles/elastic-stack/ansible-elasticsearch/defaults/main.yml#L5 .

Kind regards,

Rshad

**Pull request merged**